megaparsec-tests/megaparsec-tests.cabal

```diff
@@ -84,3 +84,6 @@ test-suite tests
 
     else
         ghc-options: -O2 -Wall
+
+    if impl(ghc >=9.8)
```

Comment:
I'd rather do without this. Even though partial functions are "bad", sometimes they are the simplest and the most reasonable solution available.

Reply:
I am not sure I understand; as far as I am aware, there are two options when using `-Werror` (as is done on CI here):

 1. Adapt usage sites of `head`/`tail` to sth else (refactorings, or other partial functions). (In this PR, this is done in one place using `NE.fromList`.)
 2. Use `-Wno-x-partial` to silence the warnings. (This is done in this PR right here.)

Are you advocating for one of these, or sth else?

Reply:
Ah, I simply misread `-Wno-x-partial` as `-Wx-partial`, missed the `no`.

Reply:
Right, would you then also prefer using `-Wno-x-partial` in order to be able to revert the `NE.fromList` change back to `head`?

Reply:
Yes, that would be nice.